realsense_camera/src/base_nodelet.cpp

```diff
@@ -231,56 +202,89 @@ namespace realsense_camera
     // print device info
     ROS_INFO_STREAM(nodelet_name_ << " - Connecting to camera with Serial No: " <<
         rs_get_device_serial(rs_device_, &rs_error_) <<
-        " USB Port ID: " << rs_get_device_usb_port_id(rs_device_, &rs_error_));
+        ", USB Port ID: " << rs_get_device_usb_port_id(rs_device_, &rs_error_));
     checkError();
-
-    // Enable streams.
-    if (enable_color_ == true)
-    {
-      enableStream(RS_STREAM_COLOR, width_[RS_STREAM_COLOR], height_[RS_STREAM_COLOR], COLOR_FORMAT, fps_[RS_STREAM_COLOR]);
-    }
-    if (enable_depth_ == true)
-    {
-      enableStream(RS_STREAM_DEPTH, width_[RS_STREAM_DEPTH], height_[RS_STREAM_DEPTH], DEPTH_FORMAT, fps_[RS_STREAM_DEPTH]);
-      enableStream(RS_STREAM_INFRARED, width_[RS_STREAM_DEPTH], height_[RS_STREAM_DEPTH], IR_FORMAT, fps_[RS_STREAM_DEPTH]);
-    }
-    getCameraOptions();
-    setStaticCameraOptions();
-
-    // Start device.
-    ROS_INFO_STREAM(nodelet_name_ << " - Starting camera");
-    rs_start_device(rs_device_, &rs_error_);
-    checkError();
-
-    is_device_started_ = true;
-
-    allocateResources();
-
     return true;
   }
 
-  void BaseNodelet::listCameras(int num_of_cameras)
+  /*
+   * List details of the detected cameras.
+   */
+  int BaseNodelet::listCameras(int num_of_cameras)
```

Comment:
Since this function is already looping through all of the devices, maybe it could return a vector of integers contains the device indexes of the validate camera type.
Then in connectToCamera() we only need to check the number of type_count, and then only loop through the camera indexes of interest; no need to check the device name.